### ISSUE TITLE
Question about training code

### ISSUE THREAD

**Issue by Gaffey** [OpenMOSS/AnyGPT#30]
Thanks for your contributions to the opensource community. There is some confusion about the training code. In `anygpt/src/stage1_pretrain.py`, I only find that the image/speech/music data is loaded, but not tokenized by the corresponding tokenizers (like SEED or SpeechTokenizer). Where do you use them to tokenize these data in pretraining?

**Comment by JunZhan2000** [comment]
Hello, the training process is as follows.

1. First, you need to use the multimodal tokenizer to discretize the image, speech, and music to obtain a token sequence.

2. Then call the function in https://github.com/OpenMOSS/AnyGPT/blob/main/anygpt/src/m_utils/anything2token.py to convert the token into a corresponding string (convenient for training with transformers).

3. Finally, as mentioned in the paper, https://github.com/OpenMOSS/AnyGPT/blob/main/anygpt/src/m_utils/prompter.py#L128 provides a function that concatenates a non-text modal content X and a text text into a complete one for training sentence.

Then write the training data into a txt or json for training

**Comment by sambalshikhar** [comment]
Can you please elaborate on the step to do the following 
"First, you need to use the multimodal tokenizer to discretize the image, speech, and music to obtain a token sequence."
As its a bit confusion where and what to run

**Comment by JunZhan2000** [comment]
> Can you please elaborate on the steps to do the following "First, you need to use the multimodal tokenizer to discretize the image, speech, and music to obtain a token sequence." As its a bit confusing where and what to run

This involves some trivial data processing code, which we have not organized. But you can find the core content in our code. For example, for how to tokenize image, you can refer to https://github.com/OpenMOSS/AnyGPT/blob/6404dbafccc10943be6bf6e24a4b99b3a6545501/anygpt/src/infer/cli_infer_chat_model.py#L100
